### ISSUE TITLE
Generate report on author self-citation

### ISSUE THREAD

**Issue by zednis** [USGCRP/gcis-ontology#114]
"Ascertain whether NCA3 authors cited their own work in their chapters, and if so, to what extent [we might not be there yet given our discussions of representation]."

**Comment by zednis** [comment]
query has 47 results.

```
PREFIX rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs:  <http://www.w3.org/2000/01/rdf-schema#>
PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
PREFIX owl:   <http://www.w3.org/2002/07/owl#>
PREFIX cito: <http://purl.org/spar/cito/>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX foaf: <http://xmlns.com/foaf/0.1/>

SELECT DISTINCT ?author as ?ContributorID str(?gn) as ?GivenName str(?ln) as ?LastName ?chapter str(?cr) as ?ChapterRole ?article str(?ar) AS ?ArticleRole
FROM <http://data.globalchange.gov>
WHERE {
  <http://data.globalchange.gov/report/nca3> gcis:hasChapter ?chapter .
  ?chapter prov:qualifiedAttribution [ prov:hadRole ?chapter_role ; prov:agent ?author ] .
  ?chapter_role rdfs:label ?cr .
  ?chapter cito:cites ?article .
  ?article prov:qualifiedAttribution [ prov:hadRole ?article_role ;prov:agent ?author ] .
  ?article_role rdfs:label ?ar .
  ?author foaf:givenName ?gn .
  ?author foaf:lastName ?ln .
}
```

**Comment by zednis** [comment]
yasgui link - http://yasgui.org/short/4J5tR83i

**Comment by justgo129** [comment]
well-done, @zednis.  @bduggan should this go to the test suite?

**Comment by zednis** [comment]
@justgo129 This query was added to gcis-sparql.  Is this ticket now ready to be closed?

**Comment by justgo129** [comment]
Assuming it works and provides the correct output (I haven't had a chance to test), yes.

**Comment by justgo129** [comment]
This is cool.  Let's add a "count" column to display the number of pubs by each nca3 author cited by his or her given chapter in order to avoid rows containing identical information.  It might be cool to also provide the URIs for each citation (e.g. Terry Mader, who authored Chapter X of the NCA3, was a co-author on publications /article/10.2567.... and /article/19.2568... , both of which are cited in that NCA3 chapter.)

**Comment by zednis** [comment]
There are no rows showing identical information.  If an author self-cites himself more than once in a given chapter, there will be a record for each citation and the title of the cited paper along with the author's contribution role to the cited paper (author/editor/etc).

Are you suggesting we remove the cited article information from the report and replace it with just a count?  Then we won't know if the self-citation is to an article the chapter author was an editor vs author on.

**Comment by justgo129** [comment]
Ah, I didn't see the rows to the right hand side of the column.  Yes, this is fine.  Closed #114.  Good job.
